### PR TITLE
[Feature] Keep the scroll position into a widget when user switches to comment mode

### LINKED ISSUE #5795: [Feature] Keep scroll position while adding comments to scrollable containers / widgets

### PR DESCRIPTION
**Pull request by pranavkanade**
## Description
This fix makes sure that we maintain the scroll position inside a widget when user switches to comment mode, if a user has scrolled into a scrollable widget (e.g. - form, table and container etc.).

Fixes #5795 

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- manual

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>

    // Code coverage diff between base branch:release and head branch: feature/keep-widget-scroll-in-comment-mode 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 51.95 **(0)** | 33.76 **(-0.01)** | 30.03 **(0.01)** | 52.6 **(0)**
 :green_circle: | app/client/src/comments/inlineComments/OverlayCommentsWrapper.tsx | 76.67 **(2.48)** | 41.67 **(4.17)** | 66.67 **(16.67)** | 77.78 **(3.71)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 50.59 **(-0.24)** | 39.41 **(-0.84)** | 36.21 **(0)** | 54.77 **(-0.27)**</details>

### CODE REVIEW

**Comment by pranavkanade** [comment]
[Link](https://github.com/appsmithorg/appsmith/issues/5795#issuecomment-884850770) to original investigation which clearly shows, `table` widget re-rendering when user switches to comment mode.

**Comment by pranavkanade** [comment]
Analysis done after the change, we can see that when user switches to comment mode only wrappers around widgets re-render and so are able to maintain the scroll position. On top of it, it'll give a performance boost (as browser will save time required for re-rendering)
Screenshot 2021-07-26 at 2 49 46 PM
Screenshot 2021-07-26 at 2 59 18 PM

File: app/client/src/comments/inlineComments/OverlayCommentsWrapper.tsx [hunk 1]
```diff
@@ -45,6 +46,7 @@ function OverlayCommentsWrapper({ children, refId, widgetType }: Props) {
 
   // create new unpublished thread
   const clickHandler = (e: any) => {
+    if (!isCommentMode) return;
```
Comment (Rishabh-Rathod) [inline app/client/src/comments/inlineComments/OverlayCommentsWrapper.tsx:49]: Inside `clickHandler` currently, this `if (!isCommentMode) return;` code is duplicate and the inner one can be removed.

**Review by Rishabh-Rathod — APPROVED** [review]
LGTM

**Review by Rishabh-Rathod — APPROVED** [review]
Looks good

**Review by rishabhsaxena — APPROVED** [review]

**Pull request merged**